R/flags.R

```diff
@@ -57,3 +57,69 @@ add_flags <- function(x) {
   }
   x
 }
+
+#' Extract flags from yspec meta data
+#' 
+#' Use `ys_flags_chr()` to get a character vector of all columns returned by 
+#' `ys_flags()`.
+#' 
+#' @param x a yspec object. 
+#' @param ... tidy-select specification of flag names to select.
+#' 
+#' @return 
+#' `ys_flags()` returns a named list of flag column names. `ys_flags_chr()`
+#' returns a character vector of all flag columns returned by `ys_flags()`.
+#' 
+#' @examples
+#' 
+#' spec <- ys_help$spec()
+#' 
+#' ys_flags(spec, nm)
+#' 
+#' ys_flags_chr(spec, nm) 
+#' 
+#' @seealso [ys_select_fl()]
+#' 
+#' @md
+#' @export
+ys_flags <- function(x, ...) {
+  flags <- pull_meta(x, "flags") 
+  if (!length(match.call(expand.dots = FALSE)$...)) {
+    return(flags)
+  }
+  Expr <- expr(c(...))
```

Comment:
Leftover assignment from debugging?

Reply:
Yes; removed.

Reply:
To expand: I was looking at using `strict = FALSE` argument to `eval_select()` and trying to report the names of available flags in case there was an error. It worked ok when the user only requested one flag name. But when the user requests one good name and one bad name, you get a list with the one good name and it's hard to tell that there was one bad name that was ignored. Going to just let `eval_select()` error out if a bad name is passed.